src/jit/morph.cpp

```diff
@@ -10223,7 +10227,11 @@ GenTree* Compiler::fgMorphBlockOperand(GenTree* tree, var_types asgType, unsigne
         if (varTypeIsSIMD(asgType))
         {
             if ((indirTree != nullptr) && (lclNode == nullptr) && (indirTree->Addr()->OperGet() == GT_ADDR) &&
-                (indirTree->Addr()->gtGetOp1()->gtOper == GT_SIMD))
+                (indirTree->Addr()->gtGetOp1()->gtOper == GT_SIMD
+#ifdef FEATURE_HW_INTRINSICS
```

Comment:
Why is this one ifdef'd, but not the others?

Also, should we have a helper method for `OperIsSimd || OperIsSimdHWIntrinsic`?

Reply:
> Why is this one ifdef'd, but not the others?

Ah, yes, we can remove ifdef.

> should we have a helper method for OperIsSimd || OperIsSimdHWIntrinsic?

Good point. Will do.